examples/mnist.py

```diff
@@ -23,15 +33,12 @@
 
 
 def _load_data(url=DATA_URL, filename=DATA_FILENAME):
-    try:
-        with gzip.open(filename, 'rb') as f:
-            data = pickle.load(f)
-    except (IOError, EOFError):
+    if not os.path.exists(filename):
```

Comment:
Hmm.  I'm not sure I understand the advantage of throwing an error (which most users won't easily understand) over just redownloading.  It seems to me that making the example as seamless as possible is desired.  Especially since I ran into that issue myself after killing the script because I thought the download was taking too long.  I agree that catching EOFError can be remove because encountering an empty file is unlikely to ever happen.  I guess it depends on if the goal is shorter code (3 fewer lines) or an example script that is less likely to break.  Also, try/except is more pythonic (apparently). ;)